### PR TITLE
Update React Router version from V5 to V6

### LINKED ISSUE #5535: React sample implementation to use updated version of React router (v.6+)
### Core Library

MSAL.js v2 (@azure/msal-browser)

### Wrapper Library

MSAL React (@azure/msal-react)

### Public or Confidential Client?

Public

### Documentation Location

https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/samples/msal-react-samples/typescript-sample

### Description

Hello all!

It would be great to have an up to date implementation sample for MSAL with react router v6+.

V6 has been out for a minute and it would be great to see a best practice implementation with this version.

Is this in the roadmap perhaps? Would be a great addition to this already nice documentation.

I'm still in the process of migrating, don't feel I got the knowledge just yet to make a PR on this topic.

Links to other sources with information on the matter is also very welcome.
Thanks a bunch in advance!

// G

### PR DESCRIPTION
**Pull request by salman90**
Update React Router version from V5 to V6 in React samples. Also update the [performance ](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dbb21fe612535bac9421d4966c88827cb445a3b8/lib/msal-react/docs/performance.md#performance)doc to reflect the new updates.

### CODE REVIEW

**Review by derisen — APPROVED** [review]
LGTM, thanks for picking this up

**Review by konstantin-msft — APPROVED** [review]

File: lib/msal-react/docs/performance.md [hunk 1]
```diff
@@ -28,19 +28,19 @@ import { NavigationClient } from "@azure/msal-browser";
  * If you would like to overwrite both you can implement INavigationClient directly instead
  */
 class CustomNavigationClient extends NavigationClient{
-    constructor(history) {
+    constructor(navigate) {
```
Comment (tnorling) [inline lib/msal-react/docs/performance.md:33]: Could you specify somewhere that this code snippet represents react-router v6 usage and perhaps link to the old commit for the v5 implementation?

**Review by tnorling — APPROVED** [review]
Just one suggestion but otherwise looks good! Thanks for doing this!

File: lib/msal-react/docs/performance.md (hunk 1 above)
Comment (salman90) [reply lib/msal-react/docs/performance.md:33]: Updated.

File: lib/msal-react/docs/performance.md [hunk 2]
```diff
@@ -18,6 +18,8 @@ If your application is doing all of the things above you can override the method
 
 Each router has their own method for doing client-side navigation and depending on how they expose the method you may need to refactor your application to support this feature. The example below will show how to implement this for `react-router-dom`. You can find a full sample apps that implement this for [react-router-dom](https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/samples/msal-react-samples/react-router-sample), [Next.js](https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/samples/msal-react-samples/nextjs-sample) and [Gatsby](https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/samples/msal-react-samples/gatsby-sample).
 
+> :information_source: This implantation uses react-router-dom version 6. If you're using react-router-dom version 5, please refer to the following: [implantation with react-router-dom version 5](https://github.com/AzureAD/microsoft-authentication-library-for-js/commit/3fe4e2e99270ad232ad1658aba83282e736cf97d#diff-a00cbe1eef40bb0fd03beb35a8e1b5580316b39187d4e96c826508ccca1b0b16).
```
Comment (derisen) [inline lib/msal-react/docs/performance.md]:
```suggestion
> :information_source: Samples mentioned above make use of `react-router-dom` v6. If you would like to use `react-router-dom` v5 instead, please refer to: [react-router-sample](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/cf3dc599adc36d1c4460bcf7d8185c9beb5283f3/samples/msal-react-samples/react-router-sample/src/App.js).
```

**Comment by AlexMachin1997** [comment]
Does this support the new data router ? At a glance it doesn't look like it does and is instead using the old approach just to get it working correctly.

**Pull request merged**